chore: bump cloud provider azure to latest version

bryan-cox opened the pull request:
**What type of PR is this?**
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind test
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:
chore: bump cloud provider azure to latest version

**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  <!-- Common commit types:
        build: Build 🏭
        chore: Maintenance 🔧
        ci: Continuous Integration 💜
        docs: Documentation 📘
        feat: Features 🌈
        fix: Bug Fixes 🐞
        perf: Performance Improvements 🚀
        refactor: Code Refactoring 💎
        revert: Revert Change ◀️
        style: Code Style 🎶
        security: Security Fix 🛡️
        test: Testing 💚 -->
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:


**Release note**:
```
none
```

bryan-cox commented:
@andyzhangx the new auth method I added in https://github.com/kubernetes-sigs/cloud-provider-azure/pull/8230 is included in sigs.k8s.io/cloud-provider-azure/pkg/azclient v0.5.0, which is what azure disk csi v1.34.1 has, but v1.34.1 doesn't have a newer cloud provider azure version where `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` was modified quite a bit. 

As a result, using v1.34.1 causing errors when authenticating with the new method I introduced because `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` is still tied to outdated code in v1.34.1.

bryan-cox commented:
Opened a similar PR in https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/2395

andyzhangx commented:
> @andyzhangx the new auth method I added in [kubernetes-sigs/cloud-provider-azure#8230](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/8230) is included in sigs.k8s.io/cloud-provider-azure/pkg/azclient v0.5.0, which is what azure disk csi v1.34.1 has, but v1.34.1 doesn't have a newer cloud provider azure version where `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` was modified quite a bit.
> 
> As a result, using v1.34.1 causing errors when authenticating with the new method I introduced because `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` is still tied to outdated code in v1.34.1.

@bryan-cox thanks, I have bumped the cloud provider azure lib to latest version to get some http client timeout setting fix.

andyzhangx commented:
/retest

bryan-cox commented:
> > @andyzhangx the new auth method I added in [kubernetes-sigs/cloud-provider-azure#8230](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/8230) is included in sigs.k8s.io/cloud-provider-azure/pkg/azclient v0.5.0, which is what azure disk csi v1.34.1 has, but v1.34.1 doesn't have a newer cloud provider azure version where `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` was modified quite a bit.
> > As a result, using v1.34.1 causing errors when authenticating with the new method I introduced because `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` is still tied to outdated code in v1.34.1.
> 
> @bryan-cox thanks, I have bumped the cloud provider azure lib to latest version to get some http client timeout setting fix.

Hey @andyzhangx I'm confused by this a bit. It looks like your commit downgraded cloud provider azure back down. I confirmed the version it was downgraded to still has the original issue so we would need the newer cloud provider version still like in the first commit.

andyzhangx commented:
> sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go

@bryan-cox I don't see there is `sigs.k8s.io/cloud-provider-azure/pkg/provider/config/azure_auth.go` change in the vendor file in this PR, can you point out where is the regression?

andyzhangx commented:
the change is already in https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/v1.31.4/vendor/sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth_conf.go#L51

andyzhangx commented:
I could see this change(https://github.com/kubernetes-sigs/cloud-provider-azure/pull/8230) has already been merged into master and release-1.31 branches.

```
 - master branch
root@andydev:~/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor# grep AADMSIDataPlaneIdentityPath ./* -R
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth_conf.go:   AADMSIDataPlaneIdentityPath string `json:"aadMSIDataPlaneIdentityPath,omitempty" yaml:"aadMSIDataPlaneIdentityPath,omitempty"`
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth.go:        if computeCredential == nil && len(config.AADMSIDataPlaneIdentityPath) > 0 {
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth.go:                computeCredential, err = dataplane.NewUserAssignedIdentityCredential(context.Background(), config.AADMSIDataPlaneIdentityPath, dataplane.WithClientOpts(azcore.ClientOptions{Cloud: clientOption.Cloud}))

# git checkout upstream/release-1.31
~/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor# grep AADMSIDataPlaneIdentityPath ./* -R
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth_conf.go:   AADMSIDataPlaneIdentityPath string `json:"aadMSIDataPlaneIdentityPath,omitempty" yaml:"aadMSIDataPlaneIdentityPath,omitempty"`
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth.go:        if computeCredential == nil && len(config.AADMSIDataPlaneIdentityPath) > 0 {
./sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth.go:                computeCredential, err = dataplane.NewUserAssignedIdentityCredential(context.Background(), config.AADMSIDataPlaneIdentityPath, dataplane.WithClientOpts(azcore.ClientOptions{Cloud: clientOption.Cloud}))
```

andyzhangx commented:
the change is also in v1.31.4: https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/v1.31.4/vendor/sigs.k8s.io/cloud-provider-azure/pkg/azclient/auth.go#L157-L161
